ls1intum/Artemis
# `Learning analytics`: Allow instructors to import competencies

### Checklist
#### General


#### Server
#### Client
- [x] I checked the course groups for displaying navigation elements (links, buttons).

### Motivation and Context


Closes #6082 

### Description

The competency management page now shows an import button, enabling instructors (and admin) to import a competency from another course, where the user is at least instructor. Importing an already existing competency or a prerequisite is not allowed.
In addition, some relevant parts of the code containing incorrect variable namings were corrected.

### Note:  Currently, the terms 'competency' and 'learning goal' are used interchangeably in Artemis. This will be fixed gradually with several small PRs, replacing all occurrences of 'learning goal' with 'competency'.

### Steps for Testing

Prerequisites:
- 2 Courses
- 1 Instructor (assigned to both courses)
- 1 Competency (belonging to only one of the courses)

1. Log in to Artemis as an Instructor
2. Go to Course Management
3. Go to the Competencies of a course
4. Click on the '+ Import Competencies' button to open the import modal
5. Competencies already belonging to the course or the prerequisites should be disabled
6. Select a competency to import
7. The selected competency should be added to the course

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage




| Class/File | Line Coverage | Comment |
|------------|-------:|-----:|
| import-component.ts | 95.83% | |
| competency-import.component.ts | 100% | |
| learning-goal-management.component.ts | 82.24% | my changes are covered |
| LearningGoalResource.java | 96% | |
| LearningGoalService.java | 100% | |

### Screenshots


<img width="1440" alt="Screenshot 2023-04-04 at 21 11 47" src="https://user-images.githubusercontent.com/22798314/229867622-7fda80a5-1212-4499-93ea-c43a396c05d7.png">
<img width="600" alt="Screenshot 2023-04-17 at 14 23 50" src="https://user-images.githubusercontent.com/22798314/232457824-1bfea092-95d0-4c4c-9ab5-cecff34f07fc.png">
